### PR TITLE
issue #40: support for h2o

### PR DESCRIPTION
**Pull request by mdancho84**
Addresses issue #40 support for the `h2o` package.

### CODE REVIEW

**Comment by mdancho84** [comment]
@thomasp85 Is it possible to have this PR reviewed and accepted? I have a blog I'm working on that uses h2o and lime. If accepted I can skip the custom functions and point readers to dev version. Thanks.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# lime 0.3.1.9000
+
+* Added support for `h2o` (@mdancho84) (#40)
```
Comment (thomasp85) [inline NEWS.md]: Please add this under 0.3.0.9999

File: R/models.R [hunk 2]
```diff
@@ -93,6 +93,20 @@ predict_model.lda <- function(x, newdata, type, ...) {
...
+predict_model.H2OModel <- function(x, newdata, type, ...){
+    if (!requireNamespace('h2o', quietly = TRUE)) {
+        stop('The h2o package is required for predicting h2o models')
+    }
+    pred <- h2o.predict(x, h2o::as.h2o(newdata))
+    h2o_model_class <- class(x)[[1]]
+    if (h2o_model_class %in% c("H2OBinomialModel", "H2OMultinomialModel")) {
+        return(as.data.frame(pred[,-1])) 
+    } else if (h2o_model_class == "H2ORegressionModel") {
+        return(as.data.frame(pred[,1]))
```
Comment (thomasp85) [inline R/models.R]: The column name for regression model output should be `'Response'`

File: R/models.R [hunk 3]
```diff
@@ -93,6 +93,20 @@ predict_model.lda <- function(x, newdata, type, ...) {
...
   )
 }
+predict_model.H2OModel <- function(x, newdata, type, ...){
+    if (!requireNamespace('h2o', quietly = TRUE)) {
+        stop('The h2o package is required for predicting h2o models')
+    }
+    pred <- h2o.predict(x, h2o::as.h2o(newdata))
+    h2o_model_class <- class(x)[[1]]
+    if (h2o_model_class %in% c("H2OBinomialModel", "H2OMultinomialModel")) {
+        return(as.data.frame(pred[,-1]))
```
Comment (thomasp85) [inline R/models.R]: Will this return a data.frame with one column per class? (I haven't used h2o myself)

File: R/models.R (hunk 2 above)
Comment (mdancho84) [reply R/models.R]: I'm making a change to the data frame column name to go from "predict" to "Regression". The regression model output will look like this after the change:
```
> predict_model(model.rf, newdata = df_h2o)
     Response
1   347.01963
2   332.42363
3   414.32455
4   373.05091
5   350.00000
6   385.49418
7   406.14417
8   415.23508
9   419.20509
10  381.11054
```

File: R/models.R (hunk 3 above)
Comment (mdancho84) [reply R/models.R]: Here is some sample output for the multiclass regression. Classes have the class probability, which is consistent with the output of `predict_model` for other classes.
```
> predict_model(model.rf, newdata = df_h2o)
    Iris.setosa Iris.versicolor Iris.virginica
1     1.0000000       0.0000000      0.0000000
2     1.0000000       0.0000000      0.0000000
3     1.0000000       0.0000000      0.0000000
4     1.0000000       0.0000000      0.0000000
5     1.0000000       0.0000000      0.0000000
6     1.0000000       0.0000000      0.0000000
7     1.0000000       0.0000000      0.0000000
8     1.0000000       0.0000000      0.0000000
9     0.8333333       0.1666667      0.0000000
10    1.0000000       0.0000000      0.0000000
```

File: NEWS.md (hunk 1 above)
Comment (mdancho84) [reply NEWS.md]: Will do.

**Comment by mdancho84** [comment]
Hey @thomasp85 wanted to see if you think the latest commit is good to go. I'm planning to release my next article over the weekend. The `lime` + `h2o` compatibility will be really helpful for the post.

**Comment by thomasp85** [comment]
I have been sick, so haven't looked at it before now

File: R/models.R (hunk 3 above)
Comment (thomasp85) [reply R/models.R]: is this based on the standard `iris` dataset? If so, the column names are wrong as they should match the class values (`setosa`, `versicolor`, `virginica`)

Comment (mdancho84) [reply R/models.R]: Yes, that's how h2o does things. I will fix.

**Comment by mdancho84** [comment]
Sorry to hear, not a problem. I'll update the column names.

File: R/models.R (hunk 3 above)
Comment (mdancho84) [reply R/models.R]: Thomas, 
I investigated more, and this is not the `iris` dataset that comes with base R. It's a different `iris` dataset that is part of `h2o`, which is why the column names are slightly different than the normal classes. To illustrate:

Here's the output of the `iris` dataset from `h2o`, and you can see the classes are different.
```
> path <- system.file("extdata", "iris.csv", package = "h2o")
> df_h2o <- h2o.importFile(path)
  |==================================================================================================| 100%
> df_h2o
   C1  C2  C3  C4          C5
1 5.1 3.5 1.4 0.2 Iris-setosa
2 4.9 3.0 1.4 0.2 Iris-setosa
3 4.7 3.2 1.3 0.2 Iris-setosa
4 4.6 3.1 1.5 0.2 Iris-setosa
5 5.0 3.6 1.4 0.2 Iris-setosa
6 5.4 3.9 1.7 0.4 Iris-setosa

[150 rows x 5 columns] 
```

Ultimately, the `predict_model` behavior should be acceptable. I ran it on the `iris` dataset that comes with base R. 
```
> lime:::predict_model.H2OModel(model.rf, newdata = iris_h2o)
  |==================================================================================================| 100%
       setosa versicolor   virginica
1   0.9989485 0.00000000 0.001051525
2   0.9989485 0.00000000 0.001051525
3   0.9989485 0.00000000 0.001051525
4   0.9989485 0.00000000 0.001051525
5   0.9989485 0.00000000 0.001051525
6   0.9989485 0.00000000 0.001051525
```

**Comment by mdancho84** [comment]
It turns out the `iris` dataset that comes with `h2o` is different, which is why we thought that the column names were getting screwed up. It turns out that when I use the `iris` dataset from base R, we get the correct column names. Not sure why they did that, but it doesn't appear that anything needs to be changed. Refer to [this comment](https://github.com/thomasp85/lime/pull/42#discussion_r144138451) for more details.

**Comment by thomasp85** [comment]
Can you resolve the conflicts - then I’ll merge

**Comment by thomasp85** [comment]
Thanks!

**Pull request merged**